openmls/src/treesync/mod.rs

```diff
@@ -193,68 +185,19 @@ impl TreeSync {
             ts_nodes.push(ts_node_option);
         }
         let tree = MlsBinaryTree::new(ts_nodes).map_err(|_| PublicTreeError::MalformedTree)?;
-        if let Some(leaf_index) = own_index_option {
-            let mut tree_sync = Self {
-                tree,
-                tree_hash: vec![],
-                own_leaf_index: leaf_index,
-            };
-            // Verify all parent hashes.
-            tree_sync
-                .verify_parent_hashes(backend, ciphersuite)
-                .map_err(|e| match e {
-                    TreeSyncParentHashError::LibraryError(e) => e.into(),
-                    TreeSyncParentHashError::InvalidParentHash => {
-                        TreeSyncFromNodesError::from(PublicTreeError::InvalidParentHash)
-                    }
-                })?;
-            // Populate tree hash caches.
-            tree_sync.populate_parent_hashes(backend, ciphersuite)?;
-            Ok(tree_sync)
-        } else {
-            debug_assert!(false, "Unable to find key package.");
-            Err(PublicTreeError::MissingKeyPackage.into())
-        }
-    }
-
-    /// Create a [`TreeSync`] instance from a vector of nodes without expecting
-    /// there to be a [`KeyPackage`] that belongs to this particular MLS client.
-    /// WARNING: Some of the [`TreeSync`] invariants will not hold for this
-    /// tree, as the `own_leaf_index` does not point to a leaf with private key
-    /// material in it.
-    pub(crate) fn from_nodes_without_leaf(
```

Comment:
We don't need a separate constructor for `without_leaf` anymore, since `TreeSync` doesn't have an `own_leaf` anymore.